js/modules/k6/http/async_request_test.go

```diff
@@ -0,0 +1,259 @@
+package http
+
+import (
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	logtest "github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/js/modulestest"
+)
+
+func wrapInAsyncLambda(input string) string {
+	// This makes it possible to use `await` freely on the "top" level
+	return "(async () => {\n " + input + "\n })()"
+}
+
+func runOnEventLoop(runtime *modulestest.Runtime, code string) error {
+	// TODO move this in modulestest.Runtime and extend it
+	err := runtime.EventLoop.Start(func() error {
+		_, err := runtime.VU.Runtime().RunString(wrapInAsyncLambda(code))
+		return err
+	})
+	runtime.EventLoop.WaitOnRegistered()
+	return err
+}
+
+func TestAsyncRequest(t *testing.T) {
+	t.Parallel()
+	t.Run("EmptyBody", func(t *testing.T) {
+		t.Parallel()
+		ts := newTestCase(t)
+
+		sr := ts.tb.Replacer.Replace
+		err := runOnEventLoop(ts.runtime, sr(`
```

Comment:
non-blocking nit: the helpers to run on the event loop inside of an async lambda make total sense. However, I would have gained a few minutes wrapping my head around it if the statement were not wrapped. I'm suggesting we unroll the statement as:
```javascript
runOnEventLoop(ts.Runtime, wrapInAsyncLambda(sr(`
```

Reply:
Hm ... I am not really certain I agree. If anything I am more for this going on top of the testCase type here and actually calling both `wrapInAsyncLambda` and `sr` automatically so you don't need to forget about it.

And then use that over the whole tests in `k6/http` . A version with `sr` should probably be extracted in `moduletest`

What in particular was not clear or confusing to you about this statement?

Reply:
Fair enough, that was really just an idea I had, but I see your point, and I'm happy to go your way 🤝 🙇🏻